### PR TITLE
Improve readibility of code

### LINKED ISSUE #1881: Minimize duplication of code in templates
**Actual Behaviour**
Currently the templates are using the same logic for search, filter etc, all the templates have duplicate code. 

**Expected Behaviour**
We should add a separate JS file to reduce the code duplication in the template files

### LINKED ISSUE #1974: Improve readibility of code
**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by shubham3211**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

Fixes #1974

### CODE REVIEW

**Comment by agbilotia1998** [comment]
@shubham3211 please check the Travis build, also we should move the code for filters to `main.js` as the logic is almost same for all. So we need to pass parameters to the procedures and minimize the duplication of code.

**Comment by shubham3211** [comment]
@agbilotia1998  I am going to make seperate pr for #1973 , I have changed the issue resolved in this pr.

File: src/backend/templates/rooms.hbs [hunk 1]
```diff
@@ -435,29 +435,29 @@
...
                       $(this).find('.session-speakers-more').text().toUpperCase().indexOf(filterVal.toUpperCase()) >= 0 ||
                       $(this).find('.event').text().toUpperCase().indexOf(filterVal.toUpperCase()) >= 0) {
-                        elemFlag = 1;
+                        elemFlag = true;
                         if(mode && temp[id] != 1) {
-                          elemFlag = 0;
+                          elemFlag = false;
                         }
                   }
                   if(elemFlag == 1) {
```
Comment (agbilotia1998) [inline src/backend/templates/rooms.hbs]: `1` should be replaced by `true`

**Comment by shubham3211** [comment]
@agbilotia1998 done :+1:

**Comment by shubham3211** [comment]
@agbilotia1998 I made the changes you asked for but it is causing the travis build to fail , similarly I was not able to make change to https://github.com/shubham3211/open-event-webapp/blob/05b84ef586b4d6c5bc2b45584b896f250f0a9d42/src/backend/templates/schedule.hbs#L450
, I have made changes to all the other places .

**Comment by agbilotia1998** [comment]
@shubham3211 you make the changes, Travis should not be affected by these. Some tests do have a problem I will sort them out.

**Comment by shubham3211** [comment]
@agbilotia1998 please review

**Pull request merged**